packages/flutter_tools/lib/src/reporting/crash_reporting.dart

```diff
@@ -26,15 +26,61 @@ const String _kStackTraceFileField = 'DartError';
 /// it must be supplied in the request.
 const String _kStackTraceFilename = 'stacktrace_file';
 
+class CrashDetails {
+  CrashDetails({
+    @required this.command,
+    @required this.error,
+    @required this.stackTrace,
+    @required this.doctorText,
+  });
+
+  String command;
+  dynamic error;
+  StackTrace stackTrace;
+  String doctorText;
+}
+
+/// Reports information about the crash to the user.
+class CrashReporter {
+  CrashReporter({
+    @required this.logger,
+    this.gitHubTemplateCreator,
+  });
+
+  final GitHubTemplateCreator gitHubTemplateCreator;
+  final Logger logger;
```

Comment:
Can you make these private?

Reply:
I did not make them private since they might be needed by the derived class.  I either could make them `@protected` or make them private with the derived class duplicating the fields.